### PR TITLE
Add docs for CRD features for 1.10

### PR DESCRIPTION
**Pull request by nikhita**
**DO NOT MERGE** until the following PRs have been merged:
- https://github.com/kubernetes/kubernetes/pull/55168 (merged)
- https://github.com/kubernetes/kubernetes/pull/58283 (merged)
- https://github.com/kubernetes/kubernetes/pull/59561 (merged)

**Preview links**:

- [Docs for subresources](https://deploy-preview-7439--kubernetes-io-vnext-staging.netlify.com/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/#subresources)
- [Custom Resources page](https://deploy-preview-7439--kubernetes-io-vnext-staging.netlify.com/docs/concepts/api-extension/custom-resources/)
- [Feature gates](https://deploy-preview-7439--kubernetes-io-vnext-staging.netlify.com/docs/reference/feature-gates/)

/cc sttts deads2k enisoc

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/website/7439)

### CODE REVIEW

**Comment by heckj** [comment]
/assign
/hold
(adding the hold for the request to validate PRs merged prior to this into 1.10 branch)

**Comment by nikhita** [comment]
@sttts Updated the docs, ptal.

Also added docs for the `Categories` API (https://github.com/kubernetes/kubernetes/pull/59561).

File: docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions.md [hunk 1]
```diff
@@ -328,6 +328,226 @@ And create it:
...
+/apis/stable.example.com/v1/namespaces/*/crontabs/status
+```
+
+and
+
+```
+/apis/stable.example.com/v1/namespaces/*/crontabs/scale
+```
+
+A custom resource can be scaled using the `kubectl scale` command.
```
Comment (nikhita) [inline docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions.md]: @sttts This is for scale. Do you think we need to include some other information for status as well?

For example, that we can use `UpdateStatus` in controllers. However, that could be more suited for client-go release notes...

File: docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions.md [hunk 2]
```diff
@@ -328,6 +328,226 @@ And create it:
...
+To enable the scale subresource, the following values are defined in the CustomResourceDefinition.
+
+1. `SpecReplicasPath` defines the JSONPath inside of a custom resource that corresponds to `Scale.Spec.Replicas`.
+
+- It is a required value.
+- Only JSONPaths under `.spec` and with the dot notation are allowed.
+- If there is no value under the `SpecReplicasPath` in the custom resource,
+the `/scale` subresource will return an error on GET.
+
+2. `StatusReplicasPath` defines the JSONPath inside of a custom resource that corresponds to `Scale.Status.Replicas`.
```
Comment (nikhita) [inline docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions.md]: @heckj Any idea why the numbers 2, 3,.. are also showing up as 1 in the webpage?

https://deploy-preview-7439--kubernetes-io-vnext-staging.netlify.com/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/#scale-subresource

Comment (heckj) [reply docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions.md]: @nikhita The jekyll markdown support for ordered lists treats the `1.` as a reference for an ordered list - and works fine when you don’t have sub-bullets, but it appears to be a limitation that you can’t mix ordered and unordered list types - for example, the bullets - or it resets the counting for the ordered lists, which is what you’re seeing

You could make them all unordered lists (getting rid of the numbers and using `-`) or all ordered, and you’d likely be fine either way

Comment (nikhita) [reply docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions.md]: @heckj Thanks! I'll make them unordered lists. :+1:

**Comment by nikhita** [comment]
All feature PRs have been merged in `kubernetes/kubernetes`. Removing the hold.

/hold cancel

**Comment by heckj** [comment]
👋 @sttts would love a review for technical accuracy, an if good, please add an `/lgtm` comment to the PR

/approve

File: docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions.md (hunk 1 above)
Comment (sttts) [reply docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions.md]: we need an example in client-go and yes, release notes. But maybe not here.

Comment (nikhita) [reply docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions.md]: :+1:

**Comment by sttts** [comment]
> would love a review for technical accuracy,

5 out of 5 :)

**Pull request merged**